File: python/412_Fizz_Buzz.py
class Solution(object):
    def fizzBuzz(self, n):
        result = []
        for i in range(1, n + 1): 
            divisibleBy5 = i % 5 == 0
            divisibleBy3 = i % 3 == 0
            
            if divisibleBy5 and divisibleBy3:
                result.append("FizzBuzz")
            elif divisibleBy5:
                result.append("Buzz")
            elif divisibleBy3:
                result.append("Fizz")
            else: 
                result.append(str(i));     
        return result

File: php/412_Fizz_Buzz.php
<?php

class Solution
{

    protected $arWords = [
        'DIVISIBLE_BY_3_5' => 'FizzBuzz',
        'DIVISIBLE_BY_3' => 'Fizz',
        'DIVISIBLE_BY_5' => 'Buzz',
    ];

    /**
     * @param Integer $n
     * @return String[]
     */
    function fizzBuzz($n)
    {
        $arResult = [];
        foreach (range(1, $n) as $number) {
            $arResult[] = $this->divisible($number);
        }

        return $arResult;
    }

    /**
     * @param Integer $n
     * @return String
     */
    protected function divisible($n)
    {
        $result = '';
        if ($n % 3 == 0 && $n % 5 == 0) {
            $result = $this->arWords['DIVISIBLE_BY_3_5'];
        } else if ($n % 5 == 0) {
            $result = $this->arWords['DIVISIBLE_BY_5'];
        } else if ($n % 3 == 0) {
            $result = $this->arWords['DIVISIBLE_BY_3'];
        } else {
            $result = (string) $n;
        }

        return $result;
    }

}
